ddns/provider/noip.py

```diff
@@ -24,12 +23,22 @@ class NoipProvider(SimpleProvider):
 
     def _validate(self):
         """
-        Validate authentication credentials for No-IP
+        Validate authentication credentials for No-IP and set up auth endpoint
         """
         if not self.id:
             raise ValueError("No-IP requires username as 'id'")
         if not self.token:
             raise ValueError("No-IP requires password as 'token'")
+        
+        # URL encode username and password to handle special characters
+        username_encoded = quote(self.id, safe="")
+        password_encoded = quote(self.token, safe="")
+        
+        # Extract domain from endpoint and create auth endpoint
+        protocol, domain = self.endpoint.split("://", 1)
+        self.auth_endpoint = "{0}://{1}:{2}@{3}".format(
```

Comment:
just update the endpoint. all the requests should with this basic auth. and need not change it in every request.

Reply:
you can add a endpoint checking before the id checking